sklearn/feature_selection/_sequential.py

```diff
@@ -190,22 +227,33 @@ def fit(self, X, y):
         current_mask = np.zeros(shape=n_features, dtype=bool)
         n_iterations = (
             self.n_features_to_select_
-            if self.direction == "forward"
+            if self.n_features_to_select == "auto" or self.direction == "forward"
             else n_features - self.n_features_to_select_
         )
+
+        old_score = -np.inf
         for _ in range(n_iterations):
-            new_feature_idx = self._get_best_new_feature(
+            new_feature_idx, new_score = self._get_best_new_feature_score(
                 cloned_estimator, X, y, current_mask
             )
+            is_auto_select = (
+                self.tol is not None and self.n_features_to_select == "auto"
+            )
+            if is_auto_select and ((new_score - old_score) < self.tol):
+                break
+
+            old_score = new_score
             current_mask[new_feature_idx] = True
 
         if self.direction == "backward":
             current_mask = ~current_mask
+
         self.support_ = current_mask
+        self.n_features_to_select_ = self.support_.sum()
 
         return self
 
-    def _get_best_new_feature(self, estimator, X, y, current_mask):
+    def _get_best_new_feature_score(self, estimator, X, y, current_mask):
         # Return the best new feature to add to the current_mask, i.e. return
```

Comment:
We should also update the docstring of this method. Could you make a proper docstring instead of the current comment?

Reply:
update docstring with https://github.com/scikit-learn/scikit-learn/pull/20145/commits/3e098b0debc7dc64ffcad3fbf3919a926d9bfeb2